### ISSUE TITLE
install failed

### ISSUE THREAD

**Issue by manin22** [balbuze/volumio-plugins#176]
Hi Balbuze,
first of all, I really appreciate your work. I would say that I have bought the Spotify Family because of you and my daughters are happy, so thanks! ;-)

I just tried to install your dev. version of volspotconnect2 on my newly reinstalled system (2.389), because the plugin in official repo wasn't still updated. Unfortunately, the installation wasn't finished and I see in the logs that there is an access right issue (/tmp/).

root@volumio:~/volspotconnect2# volumio plugin install
Compressing the plugin
Plugin succesfully compressed
Progress: 10
Status :Downloading plugin
CTRL+C
======================================================
Related Logs:
2018-06-03T05:43:02.688Z - info: Downloading plugin at http://127.0.0.1:3000/plugin-serve/volspotconnect2.zip
2018-06-03T05:43:02.734Z - info: Cannot download file http://127.0.0.1:3000/plugin-serve/volspotconnect2.zip - Error: Command failed: /bin/mv /tmp/plugins/volspotconnect2.zip /tmp/downloaded_plugin.zip
/bin/mv: cannot move '/tmp/plugins/volspotconnect2.zip' to '/tmp/downloaded_plugin.zip': Permission denied

2018-06-03T05:43:02.737Z - info: Error: Error: Error: Command failed: /bin/mv /tmp/plugins/volspotconnect2.zip /tmp/downloaded_plugin.zip
/bin/mv: cannot move '/tmp/plugins/volspotconnect2.zip' to '/tmp/downloaded_plugin.zip': Permission denied

BR, manin

**Comment by balbuze** [comment]
Did you follow exactly the instruction to install?

**Comment by manin22** [comment]
Well, not really ;-) I run it as user root (changed password).

Anyway, I wasn't successful even as volumio user (default password):
volumio@volumio:~/volspotconnect2$ volumio plugin install

This command will install the plugin on your device

Compressing the plugin
Plugin succesfully compressed
/bin/mv: cannot create regular file '/tmp/plugins/volspotconnect2.zip': Permission denied
child_process.js:524
    throw err;
    ^

Error: Command failed: /bin/mv *.zip /tmp/plugins/volspotconnect2.zip
/bin/mv: cannot create regular file '/tmp/plugins/volspotconnect2.zip': Permission denied

    at checkExecSyncError (child_process.js:481:13)
    at execSync (child_process.js:521:13)
    at install (/volumio/pluginhelper.js:671:9)
    at Object.<anonymous> (/volumio/pluginhelper.js:731:9)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)

Platform RPI2, fresh new volumio system installation, official plugin Volumio Spotify Connect2 hasn't been never installed.

**Comment by balbuze** [comment]
Sorry to be insistent, but it seems you do not apply instruction :

**Comment by balbuze** [comment]
wget https://github.com/balbuze/volumio-plugins/raw/master/plugins/music_service/volspotconnect2/volspotconnect2.zip
mkdir ./volspotconnect2
miniunzip volspotconnect2.zip -d ./volspotconnect2
cd ./volspotconnect2
volumio plugin install

**Comment by balbuze** [comment]
or your internet connection is problematic

**Comment by manin22** [comment]
Disagree, wget downloaded the file and I was able to unzip it.

Damn, how is it possible?! Tried again few minutes ago to store whole console session output, but the second attempt, exactly the same as you can see from my command history, has been successful and it works like a charm. I'm confused ;-(

root@volumio:~# history | grep volumio
   54  su - volumio
   74  su - volumio

volumio@volumio:~$ history
...
    4  wget https://github.com/balbuze/volumio-plugins/raw/master/plugins/music_service/volspotconnect2/volspotconnect2.zip
    5  mkdir ./volspotconnect2
    6  miniunzip volspotconnect2.zip -d ./volspotconnect2
    7  cd ./volspotconnect2
    8  volumio plugin install
    9  cat /var/log/volumio.log
   10  ll
   11  cd ..
   12  ll
   13  rm volspotconnect2.zip
   14  rm volspotconnect2/ -rf
   15  exit
   16  ll
   17  history
   18  wget https://github.com/balbuze/volumio-plugins/raw/master/plugins/music_service/volspotconnect2/volspotconnect2.zip
   19  mkdir ./volspotconnect2
   20  miniunzip volspotconnect2.zip -d ./volspotconnect2
   21  cd ./volspotconnect2
   22  volumio plugin install
   23  exit

Sorry for bothering and thanks for your answers. Do we close this issue ticket? Anyway, in case it would be beneficial, I can investigate if it is possible to simulate the failed procedure in bare system. Just let me know.
BR, manin

**Comment by manin22** [comment]
Summary and close notes:
the plugin installation is working properly. I have tried it again just after Volumio re-installation.

Root cause:
User mistake ;-) The issue was caused by my first failed plugin installation running as the user root. It created the temp. directory /tmp/plugins without write access to anyone else:
**drwxr-xr-x 2 root    root**      40 Jun  4 17:22 plugins
and didn't clean it at the end because of interruption. Due to that, it broke the new correct installation. So, not only the file /tmp/plugins/volspotconnect2.zip should be removed, but whole directory. After rmdir all works like a charm!

I'm sorry for this needless issue ticket. BR, manin
